File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        min_heap = []
        for n in nums:
            heapq.heappush(min_heap, n)
        count = 0
        countT = 0
        last = math.inf
        while min_heap:
            num = heapq.heappop(min_heap)
            print(num)
            if abs(num - last) <= 1 or last == math.inf:
                if num != last:
                    countT += 1
            else:
                countT = 1
            count = max(count, countT)
            last = num
        return count

        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-7.cpp
class Solution {
    public:
        int characterReplacement(string s, int k) {
                int maxFreq = 0;
                int maxRep = 0;
                int l = 0;
                std::vector<int> count(26, 0);
                for (int r = 0; r < s.size(); r++) {
                        count[s[r] - 'A']++;
                        maxFreq = std::max(maxFreq, count[s[r] - 'A']);
                        while ((r - l + 1) - maxFreq > k) {
                                count[s[l] - 'A']--;
                                l++;
                        }
                        maxRep = std::max(maxRep, r - l + 1);
                }
                return maxRep;
        }    
};
